BraveWallet/Extensions/RpcServiceExtensions.swift

```diff
@@ -15,42 +15,76 @@ extension BraveWalletJsonRpcService {
   func balance(
     for token: BraveWallet.BlockchainToken,
     in account: BraveWallet.AccountInfo,
+    decimalFormatStyle: WeiFormatter.DecimalFormatStyle = .balance,
     completion: @escaping (Double?) -> Void
   ) {
-    let convert: (String, BraveWallet.ProviderError, String) -> Void = { wei, status, _ in
-      guard status == .success && !wei.isEmpty else {
-        completion(nil)
-        return
-      }
-      let formatter = WeiFormatter(decimalFormatStyle: .balance)
-      if let valueString = formatter.decimalString(
-        for: wei.removingHexPrefix,
-        radix: .hex,
-        decimals: Int(token.decimals)
-      ) {
-        completion(Double(valueString))
-      } else {
-        completion(nil)
-      }
-    }
     network(account.coin) { [self] network in
-      if token.symbol == network.symbol {
-        balance(account.address, coin: account.coin, chainId: network.chainId, completion: convert)
-      } else if token.isErc20 {
-        erc20TokenBalance(
-          token.contractAddress(in: network),
-          address: account.address,
-          chainId: network.chainId,
-          completion: convert
+      switch account.coin {
+      case .eth:
+        ethBalance(
+          for: token,
+          in: account.address,
+          network: network,
+          completion: { wei, status, _ in
+            guard status == .success && !wei.isEmpty else {
+              completion(nil)
+              return
+            }
+            let formatter = WeiFormatter(decimalFormatStyle: decimalFormatStyle)
+            if let valueString = formatter.decimalString(
+              for: wei.removingHexPrefix,
+              radix: .hex,
+              decimals: Int(token.decimals)
+            ) {
+              completion(Double(valueString))
+            } else {
+              completion(nil)
+            }
+          }
         )
-      } else if token.isErc721 {
-        erc721TokenBalance(
-          token.contractAddress,
-          tokenId: token.tokenId,
-          accountAddress: account.address,
-          chainId: network.chainId,
-          completion: convert)
-      } else {
+      case .sol:
+        if token.symbol == network.nativeToken.symbol {
+          solanaBalance(account.address, chainId: network.chainId) { lamports, status, errorMessage in
+            guard status == .success else {
+              completion(nil)
+              return
+            }
+            let formatter = WeiFormatter(decimalFormatStyle: decimalFormatStyle)
+            if let valueString = formatter.decimalString(
```

Comment:
hmmm probably rename this formatter or at least update the method docs `value` is now a wei string in ethererum or a lamports string in solana

Reply:
Updated the function doc in [18e3530](https://github.com/brave/brave-ios/pull/5726/commits/18e35301424c8ecaa8493a0c92b6c979fc2572d1)